waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -67,52 +69,61 @@ finalizeMigration genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs onLast
 
 -- | Copies the DB migrations from the generated project dir back up to theh wasp project dir
 copyMigrationsBackToSource :: Path' Abs (Dir ProjectRootDir) -> Path' Abs (Dir DbMigrationsDir) -> IO (Either String ())
 copyMigrationsBackToSource genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs =
-  do
-    copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
-      `catch` (\e -> return $ Left $ show (e :: P.PathException))
-      `catch` (\e -> return $ Left $ show (e :: IOError))
+  copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
+    `catch` (\e -> return $ Left $ show (e :: P.PathException))
+    `catch` (\e -> return $ Left $ show (e :: IOError))
   where
-    waspMigrationsDir = SP.castDir dbMigrationsDirInWaspProjectDirAbs
-    genProjectMigrationsDir = SP.castDir $ genProjectRootDirAbs SP.</> dbRootDirInProjectRootDir SP.</> dbMigrationsDirInDbRootDir
+    waspMigrationsDir = dbMigrationsDirInWaspProjectDirAbs
+    genProjectMigrationsDir = genProjectRootDirAbs </> dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
 
 -- | This function assumes the DB schema has been generated, as it will attempt to read it from the generated code.
-writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) File' -> IO ()
+writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) (File f) -> IO ()
 writeDbSchemaChecksumToFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir = do
   dbSchemaExists <- doesFileExist dbSchemaFp
   when dbSchemaExists $ do
     checksum <- hexToString <$> checksumFromFilePath dbSchemaFp
     writeFile dbSchemaChecksumFp checksum
   where
-    dbSchemaFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaFileInProjectRootDir
-    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaChecksumInProjectRootDir
+    dbSchemaFp = genProjectRootDirAbs </> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDirAbs </> dbSchemaChecksumInProjectRootDir
 
-removeDbSchemaChecksumFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) File' -> IO ()
-removeDbSchemaChecksumFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir =
-  let dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaChecksumInProjectRootDir
-   in removeFile dbSchemaChecksumFp
+removeDbSchemaChecksumFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) (File f) -> IO ()
+removeDbSchemaChecksumFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir = removeFile dbSchemaChecksumFp
+  where
+    dbSchemaChecksumFp = genProjectRootDirAbs </> dbSchemaChecksumInProjectRootDir
+
+generatePrismaClients :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
+generatePrismaClients = liftA2 (>>) generatePrismaClientForServer generatePrismaClientForWebApp
```

Comment:
Couldn't resist :see_no_evil: 

But I can change the implementation to something normal.